src/SettingsParser.ts

```diff
@@ -0,0 +1,108 @@
+                    }
+                },
+            };
+        case visuals.FormattingComponent.ToggleSwitch:
+            return {
+                ...componentDisplayName,
+                control: {
+                    type: controlType,
+                    properties: {
+                        descriptor: { objectName: objectName, propertyName: propertyName },
+                        value: slice.value
+                    }
+                },
+
+            };
+        case visuals.FormattingComponent.NumUpDown:
+            return {
+                ...componentDisplayName,
+                control: {
+                    type: controlType,
+                    properties: {
+                        descriptor: { objectName: objectName, propertyName: propertyName },
+                        value: slice.value,
+                        // options: {
+                        //     unitSymbol: "",
+                        //     minValue: {
+                        //         value: slice.validators.min,
+                        //         type: ValidatorType.Min
+                        //     },
+                        //     maxValue: {
+                        //         value: slice.validators.max,
+                        //         type: ValidatorType.Max
+                        //     }
+                        // }
+                    }
+                },
+            };
+
+        case visuals.FormattingComponent.TextInput:
+        case visuals.FormattingComponent.TextArea:
+            return {
+                ...componentDisplayName,
+                control: {
+                    type: controlType,
+                    properties: {
+                        descriptor: { objectName: objectName, propertyName: propertyName },
+                        value: slice.value,
+                        placeholder: slice.placeholder
+                    }
+                },
+            };
+    }
+
+    return undefined;
+}
+
+
+export function getValidSelectionValue(selectionOptions: powerbi.IEnumMember[], value: any): any {
+    if (value != null && _.isNumber(value)) {
+        //It been converted to string, because when selection options bind in to the option in selectValue ng-model it been converted to a string.
+        // Then we fail to find selectValue which is the value variable here in the option values, since selectValue is a number and option values are strings.
+        let strValue = value.toString();
+        for (let option of selectionOptions) {
+            if (option.value === value) break;
+            if (option.value === strValue) return strValue;
+        }
+    }
+    return value;
+}
+
+export function getPropertyValue(value: any, defaultValue: any): any {
+    if (value && (value as powerbi.Fill).solid) {
+        return (value as powerbi.Fill).solid.color;
+    }
+
+    if (value === undefined || value === null || (typeof value === "object" && !(value as powerbi.Fill).solid)) {
```

Comment:
i think value == null should cover both cases without adding more 'true' results...